tsdb/engine/tsm1/compact.go

```diff
@@ -1060,25 +1065,30 @@ func (c *Compactor) writeNewFiles(generation, sequence int, src []string, iter K
 		logger.Debug("Compacting files", zap.Int("file_count", len(src)), zap.String("output_file", fileName))
 
 		// Write as much as possible to this file
-		err := c.write(fileName, iter, throttle, logger)
+		rollToNext, err := c.write(fileName, iter, throttle, logger)
 
-		// We've hit the max file limit and there is more to write.  Create a new file
-		// and continue.
-		if err == errMaxFileExceeded || err == ErrMaxBlocksExceeded {
+		if rollToNext {
+			// We've hit the max file limit and there is more to write.  Create a new file
+			// and continue.
 			files = append(files, fileName)
 			logger.Debug("file size or block count exceeded, opening another output file", zap.String("output_file", fileName))
 			continue
-		} else if err == ErrNoValues {
+		} else if errors.Is(err, ErrNoValues) {
 			logger.Debug("Dropping empty file", zap.String("output_file", fileName))
 			// If the file only contained tombstoned entries, then it would be a 0 length
 			// file that we can drop.
 			if err := os.RemoveAll(fileName); err != nil {
 				return nil, err
 			}
 			break
-		} else if _, ok := err.(errCompactionInProgress); ok {
-			// Don't clean up the file as another compaction is using it.  This should not happen as the
-			// planner keeps track of which files are assigned to compaction plans now.
+		} else if errors.As(err, &eInProgress) {
+			if !os.IsExist(eInProgress.err) {
```

Comment:
Didn't notice the `os.IsExist` last time. This has been deprecated for `errors.Is(err, fs.ErrExist)`. See https://pkg.go.dev/os#IsExist

Reply:
`errors.Is` gave me the willies because I was comparing a `syscall.Errno(EEXIST)` with an `fs.ErrExists`, and they have different text, so I assumed they wouldn't match.  "file exists" versus "file already exists" 

Let me make sure `errors.Is` is permissive enough.

Reply:
All fixed.